### PR TITLE
Lists are not included by default

### PR DESCRIPTION
**Pull request by alicefuzier**
For #2479 

📦 🚲

### CODE REVIEW

File: catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/works/v2/ApiV2WorksIncludesTest.scala [hunk 1]
```diff
@@ -110,16 +99,300 @@ class ApiV2WorksIncludesTest extends ApiV2WorksTestBase {
...
+        insertIntoElasticsearch(indexNameV2, itemType, work)
+
+        eventually {
+          server.httpGet(
+            path = s"/$apiPrefix/works/${work.canonicalId}?include=subjects",
+            andExpect = Status.Ok,
+            withJsonBody =
+              s"""
+                 |{
+                 |"@context": "https://localhost:8888/$apiPrefix/context.json",
```
Comment (alexwlchan) [inline catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/works/v2/ApiV2WorksIncludesTest.scala]: minor: could you clean up the indentation on this JSON please? Makes it easier to read.

File: sbt_common/display/src/main/scala/uk/ac/wellcome/display/models/v2/DisplayWorkV2.scala [hunk 2]
```diff
@@ -56,10 +56,10 @@ case class DisplayWorkV2(
   @ApiModelProperty(
     value =
       "Relates a work to the general thesaurus-based concept that describes the work's content."
-  ) subjects: List[DisplaySubject] = List(),
+  ) subjects: Option[List[DisplaySubject]] = None,
```
Comment (alexwlchan) [inline sbt_common/display/src/main/scala/uk/ac/wellcome/display/models/v2/DisplayWorkV2.scala]: Elsewhere I think we’ve had to set the dataType explicitly when we have optional lists, for example:

```scala
dataType = List[uk.ac.wellcome.display.models.v2.DisplaySubject]
```

Have you checked these show up correctly in the Swagger? (Or better, written a test for it!)

File: sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/DisplaySerialisationTestBase.scala [hunk 3]
```diff
@@ -204,6 +208,22 @@ trait DisplaySerialisationTestBase { this: Suite =>
        |}
      """.stripMargin
 
+  def contributors(c: List[Contributor[Displayable[AbstractAgent]]]) =
```
Comment (alexwlchan) [inline sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/DisplaySerialisationTestBase.scala]: nit: not a fan of single-character variable names as arguments, prefer slightly more descriptive, e.g. `con`, `prod`, `event`

Doubly so when it's a character like `l`!

File: sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/v2/DisplayWorkV2SerialisationTest.scala [hunk 4]
```diff
@@ -134,30 +115,76 @@ class DisplayWorkV2SerialisationTest
     assertJsonStringsAreEqual(actualJson, expectedJson)
   }
 
-  it("includes subject information in DisplayWorkV2 serialisation") {
+  it("includes subject information in DisplayWorkV2 serialisation with the subjects include") {
```
Comment (alexwlchan) [inline sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/v2/DisplayWorkV2SerialisationTest.scala]: Do we have tests that it ignores them without the include?

File: catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/works/v2/ApiV2WorksIncludesTest.scala [hunk 5]
```diff
@@ -110,16 +99,300 @@ class ApiV2WorksIncludesTest extends ApiV2WorksTestBase {
...
+                 |     "id": "${work.canonicalId}",
+                 |     "title": "${work.title}",
+                 |     "production": [ ${production(productionEvent)}]
+                 |   }
+          """.stripMargin
+          )
+        }
+    }
+  }
+
```
Comment (alexwlchan) [inline catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/works/v2/ApiV2WorksIncludesTest.scala]: Not sure if it's elsewhere in this file – do we have tests for work + list endpoints that if we populate all these fields, then call up the API without any includes, they don't appear in the responses?

**Comment by alexwlchan** [comment]
A couple of my comments got swallowed in the autoformatting, but I think they still apply.

File: sbt_common/display/src/main/scala/uk/ac/wellcome/display/models/v2/DisplayWorkV2.scala (hunk 2 above)
Comment (alicefuzier) [reply sbt_common/display/src/main/scala/uk/ac/wellcome/display/models/v2/DisplayWorkV2.scala]: I don't think they are needed

File: catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/works/v2/ApiV2WorksIncludesTest.scala [hunk 6]
```diff
@@ -110,16 +99,307 @@ class ApiV2WorksIncludesTest extends ApiV2WorksTestBase {
...
+        insertIntoElasticsearch(indexNameV2, itemType, work)
+
+        eventually {
+          server.httpGet(
+            path = s"/$apiPrefix/works/${work.canonicalId}?include=subjects",
+            andExpect = Status.Ok,
+            withJsonBody = s"""
+                 |{
+                 |"@context": "https://localhost:8888/$apiPrefix/context.json",
```
Comment (alexwlchan) [inline catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/works/v2/ApiV2WorksIncludesTest.scala]: nit: please tidy up these JSON examples, it makes things easier to read

Comment (alicefuzier) [reply catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/works/v2/ApiV2WorksIncludesTest.scala]: oh sorry I thought I did that

**Pull request merged**